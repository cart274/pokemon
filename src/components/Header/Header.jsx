import React from "react";
import styles from "./Header.module.css";
import logo from "../../assets/images/logo.png"
import esflag from "../../assets/images/esflag.jpg"
import enflag from "../../assets/images/enflag.jpg"
import i18n from '../../utils/i18n'
const Header = (props) =>{

  const changeLanguage = (e)=>{
    e.preventDefault();
    if(i18n.language === 'en')
      i18n.changeLanguage('es');
    else
      i18n.changeLanguage('en');
  }

  return (
    <section className={styles.header}>
      <img src={logo} alt="Logo" />
      <button className={styles.btnLang} onClick={(e)=>{changeLanguage(e)}}>
        <img  src={i18n.language === 'en' ? esflag :enflag} alt="language"/>
      </button>
    </section>
  );
}

export default Header;