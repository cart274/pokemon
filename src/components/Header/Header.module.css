@value colors: "../../assets/styles/variables.css";
@value black, cerulean, atlantis from colors;

.header{
    margin: 10px auto;
    width: 20vw;
}

.header img{
    width: 20vw;
}

.btnLang, .btnLang:active, .btnLang:focus{
    margin: -28px -20px 0 0;
    background-color: transparent;
    border: 0;
    cursor: pointer;
    outline: none;
    display: block;
    float: right;
}
.btnLang img{
    width: 25px;
    height: 16px;
}