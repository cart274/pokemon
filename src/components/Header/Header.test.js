import React from 'react';
import ReactDOM from 'react-dom';
import Header from './Header';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

describe('<Header />', () => {
    it('Renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Header />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('matches snapshots', () => {
        const wrapper = shallow(<Header/>);
        expect(wrapper).toMatchSnapshot();
    });
})
