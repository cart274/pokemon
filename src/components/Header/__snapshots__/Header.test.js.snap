// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> matches snapshot 1`] = `
<section
  className="header"
>
  <img
    alt="Logo"
    src="logo.png"
  />
  <button
    className="btnLang"
    onClick={[Function]}
  >
    <img
      alt="language"
      src="enflag.jpg"
    />
  </button>
</section>
`;

exports[`<Header /> matches snapshots 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Header />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="Logo"
          src="logo.png"
        />,
        <button
          className="btnLang"
          onClick={[Function]}
        >
          <img
            alt="language"
            src="enflag.jpg"
          />
        </button>,
      ],
      "className": "header",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Logo",
          "src": "logo.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <img
            alt="language"
            src="enflag.jpg"
          />,
          "className": "btnLang",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "language",
            "src": "enflag.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "button",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Logo"
            src="logo.png"
          />,
          <button
            className="btnLang"
            onClick={[Function]}
          >
            <img
              alt="language"
              src="enflag.jpg"
            />
          </button>,
        ],
        "className": "header",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Logo",
            "src": "logo.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              alt="language"
              src="enflag.jpg"
            />,
            "className": "btnLang",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "language",
              "src": "enflag.jpg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "button",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
